worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -123,9 +126,18 @@ public Job processJob(final String jobId) {
 
         // A mutex lock that all threads for a contract uses while writing into the shared files
         var lock = new ReentrantLock();
-        var futureResourcesHandles = patients.stream()
-                .map(patient -> patientClaimsProcessor.process(patient.getPatientId(), lock, outputFile, errorFile))
-                .collect(Collectors.toList());
+
+        var futureResourcesHandles = new ArrayList<Future<Integer>>();
+        for (PatientDTO patient : patients) {
+            var patientId = patient.getPatientId();
+
+            if (isOptOutPatient(patientId)) {
+                // This patient has opted out. Skip this patient and process the next record.
+                continue;
+            }
+
+            futureResourcesHandles.add(patientClaimsProcessor.process(patientId, lock, outputFile, errorFile));
```

Comment:
You could probably use `filter` on that stream to keep the code neat

Reply:
done